spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/dsl/task/TaskSchedule.java

```diff
@@ -37,29 +61,58 @@ public TaskSchedule(String prefix, Task task, SchedulerOperations schedulerOpera
 		this.schedulerOperations = schedulerOperations;
 	}
 
+	/**
+	 * Fluent API method to create a {@link TaskScheduleBuilder}.
+	 * @param schedulerOperations {@link SchedulerOperations} Data Flow Rest client instance.
+	 * @return A fluent style builder to create tasks.
+	 */
+	public static TaskScheduleBuilder builder(SchedulerOperations schedulerOperations) {
+		return new TaskScheduleBuilder(schedulerOperations);
+	}
+
+	/**
+	 * Schedule the task.
+	 * @param scheduleProperties Scheduling properties to use.
+	 * @param taskArgs  task arguments to pass on schedule run.
+	 */
 	public void schedule(Map<String, String> scheduleProperties, String... taskArgs) {
 		Assert.isTrue(!isScheduled(), "Task already scheduled!");
 		this.schedulerOperations.schedule(prefix, task.getTaskName(), scheduleProperties, Arrays.asList(taskArgs));
 	}
 
+	/**
+	 * Unschedule a previously scheduled task.
+	 */
 	public void unschedule() {
 		this.schedulerOperations.unschedule(getScheduleName());
 	}
 
+	/**
+	 * @return Returns true if the task is being scheduled or false otherwise.
+	 */
 	public boolean isScheduled() {
 		return this.schedulerOperations.list(this.task.getTaskName()).getContent().stream()
 				.anyMatch(sr -> sr.getScheduleName().equals(this.getScheduleName()));
 	}
 
+	/**
+	 * @return Returns the schedule properties.
+	 */
 	public Map<String, String> getScheduleProperties() {
 		Assert.isTrue(isScheduled(), "Only scheduled task can have properties");
 		return this.schedulerOperations.getSchedule(getScheduleName()).getScheduleProperties();
 	}
 
+	/**
+	 * @return Get the scheduled task.
+	 */
 	public Task getTask() {
 		return this.task;
 	}
 
+	/**
+	 * @return Get the schedule name.
+	 */
 	public String getScheduleName() {
 		return String.format("%s-scdf-%s", this.prefix, this.task.getTaskName());
```

Comment:
I ran the TaskScheduleIT Tests and they were failing because it couldn't find schedule name.   The reason is that when a user calls `getSchedulerProperties` the method verifies the taskName has an associated schedule.  The problem is that the the validation uses the `getScheduleName` method: https://github.com/spring-cloud/spring-cloud-dataflow/blob/639a64227203e4f7ec26b181ed0da9d98803f243/spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/util/task/dsl/TaskSchedule.java#L63-L65 and this method creates a compound name for the schedule that is different than what was created in the scheduling system.  In short the `isSchedule` just needs to check `sr.getScheduleName().equals(this.prefix)`.

Reply:
Ah, great catch!